src/common/core/core.js

```diff
@@ -3911,7 +3942,406 @@ define([
             }
         };
+
+        /**
+         * Renames the given pointer of the node.
+         * @param {module:Core~Node} node - the node in question.
+         * @param {string} oldName - the current name of the pointer in question.
+         * @param {string} newName - the new name of the pointer.
+         *
+         * @throws {CoreIllegalArgumentError} If some of the parameters doesn't match the input criteria.
+         * @throws {CoreIllegalOperationError} If the context of the operation is not allowed.
+         * @throws {CoreAssertError} If some internal error took place inside the core layers.
+         */
+        this.renamePointer = function (node, oldName, newName) {
+            ensureNode(node, 'node');
+            ensureType(oldName, 'oldName', 'string');
+            ensureType(newName, 'newName', 'string');
+
+            if (core.getOwnPointerPath(node, oldName) === undefined) {
+                throw new CoreIllegalOperationError('Only pointers with values can be renamed.');
+            }
+
+            return core.renamePointer(node, oldName, newName);
+        };
+
+        /**
+         * Renames the given attribute of the node.
+         * @param {module:Core~Node} node - the node in question.
+         * @param {string} oldName - the current name of the attribute in question.
+         * @param {string} newName - the new name of the attribute.
+         *
+         * @throws {CoreIllegalArgumentError} If some of the parameters doesn't match the input criteria.
+         * @throws {CoreIllegalOperationError} If the context of the operation is not allowed.
+         * @throws {CoreAssertError} If some internal error took place inside the core layers.
+         */
+        this.renameAttribute = function (node, oldName, newName) {
+            ensureNode(node, 'node');
+            ensureType(oldName, 'oldName', 'string');
+            ensureType(newName, 'newName', 'string');
+
+            return core.renameAttribute(node, oldName, newName);
+        };
+
+        /**
+         * Renames the given registry of the node.
+         * @param {module:Core~Node} node - the node in question.
+         * @param {string} oldName - the current name of the registry in question.
+         * @param {string} newName - the new name of the registry.
+         *
+         * @throws {CoreIllegalArgumentError} If some of the parameters doesn't match the input criteria.
+         * @throws {CoreIllegalOperationError} If the context of the operation is not allowed.
+         * @throws {CoreAssertError} If some internal error took place inside the core layers.
+         */
+        this.renameRegistry = function (node, oldName, newName) {
+            ensureNode(node, 'node');
+            ensureType(oldName, 'oldName', 'string');
+            ensureType(newName, 'newName', 'string');
+
+            return core.renameRegistry(node, oldName, newName);
+        };
+
+        /**
+         * Renames the given set of the node.
```

Comment:
Elaborate on what happens with inherited members and members in nodes inheriting from this.